### PR TITLE
Replace Travis by Github Action

### PR DESCRIPTION
**Pull request by pm47**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# This is a basic workflow to help you get started with Actions
+
+name: CI
```
Comment (t-bast) [inline .github/workflows/main.yml:3]:
```suggestion
name: Build & Test
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# This is a basic workflow to help you get started with Actions
```
Comment (t-bast) [inline .github/workflows/main.yml:1]: Remove? To hide the fact that we're noobs and got away with good old copy-pasting? :)

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    - name: Cache Maven dependencies
+      uses: actions/cache@v2
+      with:
+        path: ~/.m2
+        key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+        restore-keys: ${{ runner.os }}-m2
+
+    # Runs a set of commands using the runners shell
+    - name: Build with Maven
+      run: |
```
Comment (t-bast) [inline .github/workflows/main.yml:41]: Why the `|`?

File: .github/workflows/main.yml (hunk 2 above)
Comment (pm47) [reply .github/workflows/main.yml]: Done with 65b691f.

File: .github/workflows/main.yml (hunk 3 above)
Comment (pm47) [reply .github/workflows/main.yml]: This to allow multi-line commands, but I went differently with 5470640.

**Comment by pm47** [comment]
We got a test failure in [this build](https://github.com/ACINQ/eclair/runs/827108044), maybe a regression due to #1474?

> FuzzTest(true,2,0,Satoshi(372),220019009 msat,984244262 msat) -> could not setup initial htlcs: fr.acinq.eclair.channel.FeerateTooDifferent: local/remote feerates are too different: remoteFeeratePerKw=0 localFeeratePerKw=10000 (CommitmentsSpec.scala:453)

**Comment by t-bast** [comment]
> We got a test failure in this build, maybe a regression due to #1474?

Right, this is a fuzz test, I see where that's coming from, I'll fix it.

**Comment by pm47** [comment]
> just need to change the travis badge on our README

Right! good call

**Comment by pm47** [comment]
>> just need to change the travis badge on our README

> Right! good call

Done with 3bd3d01. I'd rather have used the file path method as documented [here](https://docs.github.com/en/actions/configuring-and-managing-workflows/configuring-a-workflow#example-using-a-workflow-file-path) but couldn't get it to work.

**Pull request merged**